include/openssl/kdf.h

```diff
@@ -56,41 +56,18 @@ void EVP_KDF_names_do_all(const EVP_KDF *kdf,
                           void (*fn)(const char *name, void *data),
                           void *data);
 
-# define EVP_KDF_CTRL_SET_PASS               0x01 /* unsigned char *, size_t */
-# define EVP_KDF_CTRL_SET_SALT               0x02 /* unsigned char *, size_t */
-# define EVP_KDF_CTRL_SET_ITER               0x03 /* int */
-# define EVP_KDF_CTRL_SET_MD                 0x04 /* EVP_MD * */
-# define EVP_KDF_CTRL_SET_KEY                0x05 /* unsigned char *, size_t */
-# define EVP_KDF_CTRL_SET_MAXMEM_BYTES       0x06 /* uint64_t */
-# define EVP_KDF_CTRL_SET_TLS_SECRET         0x07 /* unsigned char *, size_t */
-# define EVP_KDF_CTRL_ADD_TLS_SEED           0x08 /* unsigned char *, size_t */
-# define EVP_KDF_CTRL_RESET_HKDF_INFO        0x09
-# define EVP_KDF_CTRL_ADD_HKDF_INFO          0x0a /* unsigned char *, size_t */
-# define EVP_KDF_CTRL_SET_HKDF_MODE          0x0b /* int */
-# define EVP_KDF_CTRL_SET_SCRYPT_N           0x0c /* uint64_t */
-# define EVP_KDF_CTRL_SET_SCRYPT_R           0x0d /* uint32_t */
-# define EVP_KDF_CTRL_SET_SCRYPT_P           0x0e /* uint32_t */
-# define EVP_KDF_CTRL_SET_SSHKDF_XCGHASH     0x0f /* unsigned char *, size_t */
-# define EVP_KDF_CTRL_SET_SSHKDF_SESSION_ID  0x10 /* unsigned char *, size_t */
-# define EVP_KDF_CTRL_SET_SSHKDF_TYPE        0x11 /* int */
-# define EVP_KDF_CTRL_SET_MAC                0x12 /* EVP_MAC * */
-# define EVP_KDF_CTRL_SET_MAC_SIZE           0x13 /* size_t */
-# define EVP_KDF_CTRL_SET_SSKDF_INFO         0x14 /* unsigned char *, size_t */
-# define EVP_KDF_CTRL_SET_PBKDF2_PKCS5_MODE  0x15 /* int */
-# define EVP_KDF_CTRL_SET_UKM                0x16 /* unsigned char *, size_t */
-# define EVP_KDF_CTRL_SET_CEK_ALG            0x17 /* char * */
-# define EVP_KDF_CTRL_SET_SHARED_INFO        EVP_KDF_CTRL_SET_SSKDF_INFO
-
 # define EVP_KDF_HKDF_MODE_EXTRACT_AND_EXPAND  0
 # define EVP_KDF_HKDF_MODE_EXTRACT_ONLY        1
 # define EVP_KDF_HKDF_MODE_EXPAND_ONLY         2
 
-#define EVP_KDF_SSHKDF_TYPE_INITIAL_IV_CLI_TO_SRV     65
-#define EVP_KDF_SSHKDF_TYPE_INITIAL_IV_SRV_TO_CLI     66
-#define EVP_KDF_SSHKDF_TYPE_ENCRYPTION_KEY_CLI_TO_SRV 67
-#define EVP_KDF_SSHKDF_TYPE_ENCRYPTION_KEY_SRV_TO_CLI 68
-#define EVP_KDF_SSHKDF_TYPE_INTEGRITY_KEY_CLI_TO_SRV  69
-#define EVP_KDF_SSHKDF_TYPE_INTEGRITY_KEY_SRV_TO_CLI  70
+/* SSHKDF key exchange stages.*/
+/* See https://tools.ietf.org/html/rfc4253#section-7.2 */
+#define EVP_KDF_SSHKDF_TYPE_INITIAL_IV_CLI_TO_SRV     "A"
+#define EVP_KDF_SSHKDF_TYPE_INITIAL_IV_SRV_TO_CLI     "B"
+#define EVP_KDF_SSHKDF_TYPE_ENCRYPTION_KEY_CLI_TO_SRV "C"
+#define EVP_KDF_SSHKDF_TYPE_ENCRYPTION_KEY_SRV_TO_CLI "D"
+#define EVP_KDF_SSHKDF_TYPE_INTEGRITY_KEY_CLI_TO_SRV  "E"
+#define EVP_KDF_SSHKDF_TYPE_INTEGRITY_KEY_SRV_TO_CLI  "F"
```

Comment:
This is wrong. The "A" etc. should have been just a comment. Otherwise on the EBCDIC machines this will be incorrect.

Reply:
IMO it would be acceptable to use just {65,0} (without the cast). Then of course you need to adjust the description in the manpage. Something like `Char array initializer of value {65, 0}.`

Reply:
As I said above, the alternative is `{ (char)65, (char)0 }`, ...  That should work on all platforms and is a valid C90 array initializer (up to my knowledge).  Any opinions?

Reply:
I would be ok with that.

Reply:
Done.